### DISCUSSION TITLE
I have a question about jwt and string format

### DISCUSSION THREAD

**Discussion by GwonMinYong** [pocketbase/pocketbase#3392] [Q&A] [answered]
1. Is there a way to modify Auth fields without using email?
2. How can I customize the payload value of jwt that I create with 'Auth with password'?
3. I want to set the string format to utf-8 and put in a language value other than English, what should I do?

**Comment by ganigeorgiev** [comment]
Without any context the questions doesn't make much sense to me and I'm not sure how to answer them, but I'll try:

> 1. Is there a way to modify Auth fields without using email?

I don't understand what do you mean. If you want to perform a user update you don't need an email, just the user id (the same as for the regular records).

> 2. How can I customize the payload value of jwt that I create with 'Auth with password'?

We store in the JWT payload only the minimal set of record data that is public safe and doesn't change.
You can't add other fields to the payload, unless you create your own token(s) similar to [tokens.NewRecordAuthToken](https://github.com/pocketbase/pocketbase/blob/469769d270250e498e73391b518f28fe2cad4b8c/tokens/record.go#L13-L26), but again I don't understand why would you want to do that.

> 3. I want to set the string format to utf-8 and put in a language value other than English, what should I do?

Go string values are UTF-8 by default.


Could you elaborate on your use case and what you are trying to do?

**Reply by GwonMinYong** [reply]
I'm sorry my question was out of context. And thank you very much for your kind and quick reply.
'1.' was a question that happened by my mistake. I'm sorry.
'2.' I wanted to put the desired data among ‘Schema fields’ into the JWT. And when reading a JWT stored in a cookie, there was data I wanted to read without API communication.
Of course I know there are various solutions without adding to the JWT.
Thank you for your kind reply.
'3.' In the case of,
Since I'm Korean, I wanted to add Korean to 'username'. But ' "message": "Must be in a valid format." ' I received this message and mistakenly thought it was a problem with 'String format'.
In conclusion, what I want to do is to save Hangul in the 'username' and the ‘Schema fields’ I created.
Lastly, I apologize for my insufficient questions and thank you for your kind answers.

**Reply by ganigeorgiev** [reply, marked as answer]
The `username` field options and the accepted characters are not customizable at the moment.
The `username` must satisfy the regex: `^[\w][\w\.]*$`.

**Reply by GwonMinYong** [reply]
Thank you for your kind reply. It was a great help.
